tests/test_director.py

```diff
@@ -299,6 +329,29 @@ def test_05_register_ecu_serial(self):
     self.assertIn(secondary_serial, inventory.ecu_public_keys)
     self.assertEqual(
         keys_pub['secondary'], inventory.ecu_public_keys[secondary_serial])
+    self.assertEqual(
+        keys_pub['secondary'], inventory.get_ecu_public_key(secondary_serial))
+
+    # This should be empty, but should not raise an UnknownECU error now.
+    self.assertFalse(inventory.get_ecu_manifests(secondary_serial))
+    self.assertIsNone(inventory.get_last_ecu_manifest(secondary_serial))
+
+    # Try registering the Secondary again. Normally, we expect a Spoofing error
```

Comment:
Try to break up some of these lines, easier to read IMO.  You might have chunked all this together to make it a logical unit, but I found it difficult to eyeball.